datadog/resource_datadog_dashboard.go

```diff
@@ -995,9 +995,106 @@ func getNonGroupWidgetSchema() map[string]*schema.Schema {
 				Schema: getRunWorkflowDefinitionSchema(),
 			},
 		},
+		"split_graph_definition": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			MaxItems:    1,
+			Description: "The definition for a Split Graph widget.",
+			Elem: &schema.Resource{
+				Schema: getSplitGraphDefinitionSchema(),
+			},
+		},
 	}
 }
 
+func getSplitGraphSourceWidgetSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
```

Comment:
I was able to create a terraform plan and apply without any widgets defined under `source_widget_definition`. This was the error thrown:
```
│ Error: error updating dashboard: json: error calling MarshalJSON for type *datadogV1.Dashboard: json: error calling MarshalJSON for type datadogV1.Widget: json: error calling MarshalJSON for type datadogV1.WidgetDefinition: json: error calling MarshalJSON for type datadogV1.SplitGraphWidgetDefinition: json: error calling MarshalJSON for type datadogV1.SplitGraphSourceWidgetDefinition: unexpected character ''
```

Is this something we can catch before runtime? It's related to my other feedback on the PR.

To repro:
```
resource "datadog_dashboard" "hello" {
  title       = "Dasha Testing Split Graphs"
  description = "A Datadog Dashboard for the deployment"
  layout_type = "ordered"
  widget {
    widget_layout {
      x      = 0
      y      = 0
      width  = 12
      height = 5
    }
    split_graph_definition {
      title = "hello"
      source_widget_definition{
      }
      split_config{
        split_dimensions{
          one_graph_per = "service"
        }
        limit = 24
        sort{
          compute{
            aggregation = "sum"
            metric = "system.cpu.user"
          }
          order = "desc"
        }
        static_splits{
          split_vector{
            tag_key = "service"
            tag_values = ["cassandra"]
          }
          split_vector{
            tag_key = "datacenter"
            tag_values = []
          }
        }
        static_splits{
          split_vector{
            tag_key = "demo"
            tag_values = ["env"]
          }
        }
      }
      size = "md"
      has_uniform_y_axes = true
      live_span = "5m"
    }
  }

}
```

Reply:
What terraform cli version and commit are you on? The error you're getting is something I thought was fixed with [55bf070](https://github.com/DataDog/terraform-provider-datadog/pull/2129/commits/55bf070acbcd7b2ed7a454e4fcb3b46159074abf), but when I try your repro config locally I get a different error for line 7972, where `terraformSourceWidget[0]` is nil and panics on cast.

Reply:
https://github.com/DataDog/terraform-provider-datadog/pull/2129/commits/08a052939927d80951623679aef29b6e01381fd0

Reply:
fixed in here https://github.com/DataDog/terraform-provider-datadog/pull/2129/commits/4edd5c20624d2d60ad5f30380e8cf47829247fb6